### PR TITLE
Move type relations into submodule `relate` in rustc_infer, and notify when it has changed

### PR DESCRIPTION
**Pull request by compiler-errors**
r? lcnr

### CODE REVIEW

File: triagebot.toml [hunk 1]
```diff
@@ -441,6 +441,14 @@ cc = ["@rust-lang/miri"]
 message = "Some changes occurred in need_type_info.rs"
 cc = ["@lcnr"]
 
+[mentions."compiler/rustc_middle/src/ty/relate.rs"]
+message = "Type relation code was changed"
```
Comment (compiler-errors) [inline triagebot.toml:445]: idk, should this message be scarier?

File: compiler/rustc_infer/src/infer/relate/mod.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+//! This module contains the definitions of most `TypeRelation`s in the type system
+//! (except for the NLL `TypeRelating`, and some relations used for diagnostics
```
Comment (lcnr) [inline compiler/rustc_infer/src/infer/relate/mod.rs:2]: why did you not move `nll_relate` as well? I agree that diagnostics and heuristics don't necessarily have to be moved, but I do think `nll_relate` should be.

**Review by lcnr — COMMENTED** [review]
happy with this PR, r=me after fixing CI

File: compiler/rustc_infer/src/infer/relate/mod.rs (hunk 2 above)
Comment (compiler-errors) [reply compiler/rustc_infer/src/infer/relate/mod.rs:2]: I'll move nll_relate too

Comment (lcnr) [reply compiler/rustc_infer/src/infer/relate/mod.rs:2]: the comment is outdated now :grin:

File: triagebot.toml (hunk 1 above)
Comment (lcnr) [reply triagebot.toml:445]: I think it's fine, I trust us to look at the PR and check even without a clear message

**Comment by compiler-errors** [comment]
ok finally
@bors r=lcnr

**Pull request merged**